clients/js-legacy/src/actions.ts

```diff
@@ -0,0 +1,283 @@
+import type { ConfirmOptions, Connection, Signer, TransactionSignature } from '@solana/web3.js';
+import { PublicKey, sendAndConfirmTransaction, SystemProgram, Transaction } from '@solana/web3.js';
+import {
+    createCloseContextStateInstruction,
+    createVerifyCiphertextCiphertextEqualityInstruction,
+    createVerifyCiphertextCommitmentEqualityInstruction,
+    createVerifyPubkeyValidityInstruction,
+    createVerifyZeroCiphertextInstruction,
+} from './instructions';
+import {
+    CIPHERTEXT_CIPHERTEXT_EQUALITY_CONTEXT_ACCOUNT_SIZE,
+    CIPHERTEXT_COMMITMENT_EQUALITY_CONTEXT_ACCOUNT_SIZE,
+    PUBKEY_VALIDITY_CONTEXT_ACCOUNT_SIZE,
+    ZERO_CIPHERTEXT_CONTEXT_ACCOUNT_SIZE,
+    ZK_ELGAMAL_PROOF_PROGRAM_ID
+} from './constants';
+import {
+    ElGamalCiphertext,
+    ElGamalKeypair,
+    ElGamalPubkey,
+    PedersenCommitment,
+    PedersenOpening
+} from '@solana/zk-sdk';
+
+/** Context state account information to be used as parameters to functions */
+export interface contextStateInfo {
+    /**
+     * Keypair of the context state account. If provided, use the system
+     * program to create the context state account.
+     */
+    keypair?: Signer,
+    /** Address of the context state account */
+    address: PublicKey,
```

Comment:
Rather than having these two separate fields, how about making an `account` field as a `(Signer | PublicKey)`, similar to how we treat multisig participants in the spl-token JS client. Then you can do a `account instanceof PublicKey ?` checks where needed

Reply:
Oh yeah this is much simpler! Updated in [01378be](https://github.com/solana-program/zk-elgamal-proof/pull/1/commits/01378bec7bb2a34fdbd0a625153e6eb9d6f5a190).